src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```diff
@@ -0,0 +1,755 @@
+  const boost::property_tree::ptree& available_shells = _ptDevice.get_child("platform.available_shells");
+
+  boost::property_tree::ptree platform_to_flash;
+  for (auto& image : available_shells) {
+    if ((image.second.get<std::string>("vbnv")).compare(vbnv) == 0) {
+      platform_to_flash = image.second;
+      break;
+    }
+  }
+  std::pair<std::string, std::string> s = deployment_path_and_filename(platform_to_flash.get<std::string>("file"));
+  std::cout << boost::format("  %-20s : %s\n") % "Deployment File" % s.first;
+  std::cout << boost::format("  %-20s : %s\n") % "Deployment Directory" % s.second;
+  std::cout << boost::format("  %-20s : %s\n") % "Size" % file_size(platform_to_flash.get<std::string>("file").c_str());
+  std::cout << boost::format("  %-20s : %s\n\n") % "Timestamp" % get_file_timestamp(platform_to_flash.get<std::string>("file").c_str());
+
+  std::cout << boost::format("  %-20s : %s\n") % "Platform" % platform_to_flash.get<std::string>("vbnv", "N/A");
+  std::cout << boost::format("  %-20s : %s\n") % "SC Version" % platform_to_flash.get<std::string>("sc_version", "N/A");
+  auto logic_uuid = platform_to_flash.get<std::string>("logic-uuid", "");
+  if (!logic_uuid.empty()) {
+    std::cout << boost::format("  %-20s : %s\n") % "Platform UUID" % logic_uuid;
+  } else {
+    std::cout << boost::format("  %-20s : %s\n") % "Platform ID" % platform_to_flash.get<std::string>("id", "N/A");
+  }
+}
+
+static void
+report_status(const std::string& vbnv, boost::property_tree::ptree& pt_device)
+{
+  std::cout << "----------------------------------------------------\n";
+  pretty_print_platform_info(pt_device, vbnv);
+  std::cout << "----------------------------------------------------\n";
+
+  std::stringstream action_list;
+
+  if (!pt_device.get<bool>("platform.status.shell"))
+    action_list << boost::format("  [%s] : Program base (FLASH) image\n") % pt_device.get<std::string>("platform.bdf");
+
+  if (!pt_device.get<bool>("platform.status.sc") && !pt_device.get<bool>("platform.status.is_factory") && !pt_device.get<bool>("platform.status.is_recovery"))
+    action_list << boost::format("  [%s] : Program Satellite Controller (SC) image\n") % pt_device.get<std::string>("platform.bdf");
+
+  if (!action_list.str().empty()) {
+    std::cout << "Actions to perform:\n" << action_list.str();
+    std::cout << "----------------------------------------------------\n";
+  }
+}
+
+static bool
+are_shells_equal(const DSAInfo& candidate, const DSAInfo& current)
+{
+  if (current.dsaname().empty())
+    throw std::runtime_error("Current shell name is empty.");
+
+  return ((candidate.dsaname() == current.dsaname()) && candidate.matchId(current));
+}
+
+static bool
+are_scs_equal(const DSAInfo& candidate, const DSAInfo& current)
+{
+  if (current.dsaname().empty())
+    throw std::runtime_error("Current shell name is empty.");
+
+  return ((current.bmcVer.compare("INACTIVE") == 0) || (candidate.bmc_ver() == current.bmc_ver()));
```

Comment:
A lot is going on here.  Please refactor to check 1 DRC at a time with an early exit.